2024/YoctoProjectDeveloperDay/practical-sec-tee.md

```diff
@@ -0,0 +1,994 @@
+        - TAs (like crypto or PKCS#11);
+        - Userspace clients (like OpenSSL), libs (at least for communicating
+          with TEE) and tools (TEE-specific tools, like tee-supplicant for
+          OP-TEE);
+        - Linux Kernel drivers and devicetree configuration.
+    - Mention U-Boot integration as well.
+
+---
+
+# Compilation from source
+
+.left-code-column50[
+```bash
+meta-arm
+(...)
+├── recipes-bsp
+(...)
+│   ├── trusted-firmware-a
+(...)
+│   │   ├── tf-a-tests_2.10.0.bb
+│   │   ├── trusted-firmware-a_2.10.3.bb
+│   │   └── trusted-firmware-a.inc
+│   ├── trusted-firmware-m
+│   │   ├── trusted-firmware-m_2.0.0.bb
+│   │   ├── trusted-firmware-m-2.0.0-src.inc
+│   │   ├── trusted-firmware-m.inc
+(...)
+├── recipes-security
+│   ├── optee
+│   │   ├── optee-client
+│   │   │   ├── tee-supplicant@.service
+│   │   │   └── tee-supplicant.sh
+│   │   ├── optee-client_4.1.0.bb
+│   │   ├── optee-client.inc
+│   │   ├── optee-examples_4.1.0.bb
+│   │   ├── optee.inc
+(...)
+│   │   ├── optee-os_4.1.0.bb
+│   │   ├── optee-os.inc
+(...)
+```
+]
+.right-code-column50[
+```bash
+meta-arm-bsp
+(...)
+├── recipes-bsp
+(...)
+│   ├── trusted-firmware-a
+(...)
+│   │   ├── fiptool-native_2.8.6.bb
+│   │   ├── tf-a-tests_2.8.0.bb
+│   │   ├── tf-a-tests_%.bbappend
+│   │   ├── trusted-firmware-a_%.bbappend
+(...)
+│   ├── trusted-firmware-m
+(...)
+│   │   ├── trusted-firmware-m_%.bbappend
+│   │   └── trusted-firmware-m-corstone1000.inc
+(...)
+├── recipes-security
+│   ├── optee
+(...)
+│   │   ├── optee-client
+│   │   │   ├── tee-supplicant@.service
+│   │   │   └── tee-supplicant.sh
+(...)
+│   │   ├── optee-ftpm_%.bbappend
+│   │   ├── optee-os_4.%.bbappend
+(...)
+```
+]
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: To show that meta-arm has everithing neccessary and
+  even more.
+- What to say:
+    - TODO
+- Notes:
+    - Porting problems:
+        - TODO
+
+---
+
+# Using vendor binaries
+
+.center[ <img src="./img/vendor-binaries-meme.png" height="350px"> ]
+
+.center[ <span style="font-size:30px;">There's some kind of mess here!</span>]
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: To say that though vendor supplied binaries were
+  sopposed to make development easier, they made it even worse.
+- What to say:
+    - TODO
+- Notes:
+    - Weak points:
+        - You have to trust vendor binaries;
+        - You have to use vendor-specific tools, that are often not suitable for
+          your environment, are very outdated, and lack support;
+        - You do not have source code, so, debugging becames nearly impossible;
+        - If you have a small contract or do not have direct contact with
+          vendor, you will have to wait for support or, if you do not have time,
+          you will have to go for an endless web-digging.
+
+---
+
+# U-Boot integration
+
+1. Turn on `CONFIG_TEE` and `CONFIG_OPTEE` (and related configs, if needed)
+  configs.
+2. Link ATF (`BL31`):
```

Comment:
What does BL31 stand for?

Reply:
`BL1`: Boot ROM;
`BL2`: SPL and TPL;
`BL3`: Bootloader (U-Boot) with integrated ATF and OP-TEE;
`BL31`: ATF (so, it is explicitly sown that it is a part of U-Boot image);
`BL32`: OP-TEE (so, it is explicitly sown that it is a part of U-Boot image);
`BL33`: REE bootloader (U-Boot).

More about this [here](https://trustedfirmware-a.readthedocs.io/en/latest/getting_started/image-terminology.html).

Reply:
I guess it can be summed up in 1-2 sentences. Could you add notes what you want to say about it? You know, in the case the connection drops so I could present your part of the presentation.